functionary/ui/views/packages.py

```diff
@@ -0,0 +1,23 @@
+from django.views.generic.detail import DetailView
+from django.views.generic.list import ListView
+
+from core.models import Function, Package
+
+
+class PackageListView(ListView):
+    model = Package
+
+    def get_queryset(self):
+        """Filter packages not in the specified environment."""
+
+        env_id = self.request.session["environment_id"]
+        return super().get_queryset().filter(environment__id=env_id)
+
+
+class PackageDetailView(DetailView):
+    model = Package
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["functions"] = Function.objects.filter(package=self.get_object())
```

Comment:
You can leverage the django relationships here to get the available functions through the package itself, rather than having to import the `Function` model:

```python
context["functions"] = self.get_object().function_set.all()
```

If you go back to the `Package` model and set `related_name="functions"` on the `package` field, you can do a slightly more natural version of that:

```python
context["functions"] = self.get_object().functions.all()
```